### PR TITLE
Issue #7832: Publish pitest report as artifact at circleci

### PR DESCRIPTION
**Pull request by gaurabdg**
Fixes Issue #7832 
Please ignore, because I am testing the CI.

### CODE REVIEW

**Comment by gaurabdg** [comment]
Please ignore as I am testing the CI

**Comment by romani** [comment]
@gaurabdg , to avoid shaking all CIs while you debuging on remote, please add `RUN_JOB=0` to all CIs configs except for circleci.
in CircleCI config please comment out all https://github.com/checkstyle/checkstyle/blob/master/.circleci/config.yml#L230 except for one.  It will significantly speedup process.

File: .circleci/config.yml [hunk 1]
```diff
@@ -67,9 +67,13 @@ jobs:
               CMD="$(circleci tests split commands.txt)"
               echo "Command: $CMD"
               eval $CMD
+              mkdir pitest-report && mv ./target/pit-reports/*/* pitest-report
+              ls -R | grep ":$" | sed -e 's/:$//' -e 's/[^-][^\/]*\//--/g' -e 's/^/   /' -e 's/-/|/'
             else
               echo "build is skipped ..."
             fi
+      - store_artifacts:
+          path: ./pitest-report
```
Comment (romani) [inline .circleci/config.yml]: I would make it simple:
```
✔ ~/java/github/checkstyle/checkstyle [master|✔] 
$ ls -la target/
total 32
drwxrwxr-x  8 rivanov rivanov 4096 Mar 13 06:59 .
drwxrwxr-x 13 rivanov rivanov 4096 Mar 13 06:57 ..
-rw-rw-r--  1 rivanov rivanov    0 Mar 13 06:59 actual.txt
drwxrwxr-x  3 rivanov rivanov 4096 Mar 13 06:57 classes
drwxrwxr-x  4 rivanov rivanov 4096 Mar 13 06:57 generated-sources
drwxrwxr-x  3 rivanov rivanov 4096 Mar 13 06:57 generated-test-sources
-rw-rw-r--  1 rivanov rivanov    0 Mar 13 06:59 ignored.txt
drwxrwxr-x  4 rivanov rivanov 4096 Mar 13 06:57 maven-status
drwxrwxr-x  3 rivanov rivanov 4096 Mar 13 06:58 pit-reports
drwxrwxr-x  4 rivanov rivanov 4096 Mar 13 06:57 test-classes
✔ ~/java/github/checkstyle/checkstyle [master|✔] 
 $ ls -la target/pit-reports/
total 12
drwxrwxr-x 3 rivanov rivanov 4096 Mar 13 06:58 .
drwxrwxr-x 8 rivanov rivanov 4096 Mar 13 06:59 ..
drwxrwxr-x 3 rivanov rivanov 4096 Mar 13 06:59 202003130658
✔ ~/java/github/checkstyle/checkstyle [master|✔] 
$ ls -la target/pit-reports/202003130658/
total 24
drwxrwxr-x 3 rivanov rivanov 4096 Mar 13 06:59 .
drwxrwxr-x 3 rivanov rivanov 4096 Mar 13 06:58 ..
drwxrwxr-x 2 rivanov rivanov 4096 Mar 13 06:59 com.puppycrawl.tools.checkstyle.checks
-rw-rw-r-- 1 rivanov rivanov 1634 Mar 13 06:59 index.html
-rw-rw-r-- 1 rivanov rivanov 5514 Mar 13 06:59 style.css
```

just let circleci to copy whole `pit-reports` folder, there will be always one report.
such artifacts are for maintainers, they know this structure, and will have no problems.
New contributors will never find artifacts, and they will test in on their local, contributors do not need this report that much.

Comment (romani) [reply .circleci/config.yml]: or at least keep all files under `target` folder, to keep repo sources clean (it is very good practice, we try to fix all CIs to not put anything to repo to have git status clean after any executions).

Comment (gaurabdg) [reply .circleci/config.yml]: > just let circleci to copy whole `target/pit-reports/` folder, there will be always one report.
> such artifacts are for maintainers, they know this structure, and will have no problems.
> New contributors will never find artifacts, and they will test in on their local, contributors do not need this report that much.

Oh okay, I was trying to do this(simplifying for user). I'll upload the whole folder.

**Comment by romani** [comment]
looks good - https://40329-12499251-gh.circle-artifacts.com/0/target/pit-reports/202003131925/com.puppycrawl.tools.checkstyle.ant/CheckstyleAntTask.java.html#grouporg.pitest.mutationtest.report.html.SourceFile@29bbb7cf_291

please enable all CIs back, prepare for PR for merge.

**Comment by gaurabdg** [comment]
done

**Comment by strkkk** [comment]
@romani I see that report is stored during CI execution, but I don't understand where it published.
In case I need to find these reports, how can I get them?

**Comment by rnveach** [comment]
@strkkk Click on details next to circle CI.
https://circleci.com/gh/checkstyle/checkstyle/40423?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link
Click on Artifacts tab.
https://circleci.com/gh/checkstyle/checkstyle/40423#artifacts/containers/0
Right below "Test Summary" you will see "v Container 0". This is a folder listing for the CI and you will see "target" and "pit-reports". Keep expanding each one and you will start to see files and they will have links on them. Click on one of the links and it will take you to:
https://40423-12499251-gh.circle-artifacts.com/0/target/pit-reports/202003140038/com.puppycrawl.tools.checkstyle/index.html

Please note each Circle CI run can have multiple containers in it.

**Comment by romani** [comment]
> Should we only publish report if CI fails?

it is better to publish always, as it takes only 1second. There might be cases where we need to report even all is green as there might be problem is final exit code generation (reminder: we wrap pitest with heavy shell script)

**Comment by romani** [comment]
@strkkk  and @pbludov , I merged this PR as I need to asap in  https://github.com/checkstyle/checkstyle/pull/7836#discussion_r392541491 .

But please be familiar with this feature, pin me at any time if something is still not clear.

**Pull request merged**